### PR TITLE
Convert Entropy to build with dotnet and update samples

### PR DESCRIPTION
**Pull request by JunTaoLuo**
#90 

Known issues:
- [x] Identity insert, affects Data.SqlServer
- [x] Owin/Nowin 
- [ ] CLI issue - https://github.com/dotnet/cli/issues/1304, affects Localization.StarterWeb and MusicStore.Spa, can be worked around by running `dotnet run -f dnxcore50`
- [x] CLI issue - https://github.com/dotnet/cli/issues/1305, affects Localization.StarterWeb, fixed but we need to update the cli on our feed
- [ ] CLI issue - https://github.com/dotnet/cli/issues/1394, affects Localization.StarterWeb, Diagnostics.StatusCodes.Mvc

### CODE REVIEW

**Comment by JunTaoLuo** [comment]
cc @Tratcher, There are still two samples I couldn't run due to blocking cli issue but the rest of the samples have been updated.

**Comment by JunTaoLuo** [comment]
cc @victorhurdugaci

File: samples/Config.Providers.Web/project.json [hunk 1]
```diff
@@ -16,5 +16,8 @@
   },
   "commands": {
```
Comment (victorhurdugaci) [inline samples/Config.Providers.Web/project.json]: Commands are gone

File: samples/Config.WalkingValues.Web/project.json [hunk 2]
```diff
@@ -17,5 +17,8 @@
   },
   "commands": {
```
Comment (victorhurdugaci) [inline samples/Config.WalkingValues.Web/project.json]: commands are gone

**Comment by victorhurdugaci** [comment]
Don't you need to update the build.cmd/.sh files?

**Comment by JunTaoLuo** [comment]
This is just to fix samples, I am doing the switch to dotnet in a separate commit.

File: samples/Config.Providers.Web/project.json (hunk 1 above)
Comment (JunTaoLuo) [inline samples/Config.Providers.Web/project.json]: I could but I'm under the impressions that most people are still using dnx/dnu and it's probably best to preserve the ability to run these samples for now until most people have switched to dnx/dnu? For one, I'm not sure if VS tooling has caught up to using dotnet to run samples.

Comment (Tratcher) [inline samples/Config.Providers.Web/project.json]: Yes, leave them until we get new vs tooling

File: test/FunctionalTests/BuilderTests/BuilderFilteringWebTests.cs [hunk 3]
```diff
@@ -20,13 +20,13 @@ public class BuilderFilteringWebTests
         [ConditionalTheory]
         [OSSkipCondition(OperatingSystems.Linux)]
         [OSSkipCondition(OperatingSystems.MacOSX)]
-        [InlineData(ServerType.WebListener, RuntimeFlavor.Clr, RuntimeArchitecture.x86, "http://localhost:5501")]
+        //[InlineData(ServerType.WebListener, RuntimeFlavor.Clr, RuntimeArchitecture.x86, "http://localhost:5501")]
```
Comment (Tratcher) [inline test/FunctionalTests/BuilderTests/BuilderFilteringWebTests.cs]: Why are these disabled?

Comment (victorhurdugaci) [inline test/FunctionalTests/BuilderTests/BuilderFilteringWebTests.cs]: They're 32 bit tests

File: test/FunctionalTests/Diagnostics/DiagnosticsStatusCodesMvcTests.cs [hunk 4]
```diff
@@ -151,7 +151,7 @@ public class DiagnosticsStatusCodesMvcTests
 
         [ConditionalTheory]
         [OSSkipCondition(OperatingSystems.Windows)]
-        [InlineData(ServerType.Kestrel, RuntimeFlavor.Mono, RuntimeArchitecture.x64, "http://localhost:6129")]
+        [InlineData(ServerType.Kestrel, RuntimeFlavor.CoreClr, RuntimeArchitecture.x64, "http://localhost:6129")]
```
Comment (Tratcher) [inline test/FunctionalTests/Diagnostics/DiagnosticsStatusCodesMvcTests.cs]: Shouldn't these be Clr, not CoreClr? We still want them to run on mono.

**Comment by JunTaoLuo** [comment]
:up::date:

**Comment by JunTaoLuo** [comment]
@victorhurdugaci @BrennanConroy @Tratcher please take a look, I'd like to get this in tomorrow

**Comment by Tratcher** [comment]
Looks boring. :shipit:

**Comment by BrennanConroy** [comment]
Agreed, assuming you ran locally :shipit:

**Comment by JunTaoLuo** [comment]
Yup, though 8 tests will fail due to the issues mentioned above. Will merge shortly.

**Pull request merged**